src/main/java/com/powsybl/openloadflow/network/VoltageControl.java

```diff
@@ -120,46 +137,69 @@ private static void addVoltageControls(List<VoltageControl<?>> voltageControls,
         List<VoltageControl<?>> voltageControls = new ArrayList<>();
         LfZeroImpedanceNetwork zn = bus.getZeroImpedanceNetwork(LoadFlowModel.AC);
         if (zn != null) { // bus is part of a zero impedance graph
             for (LfBus zb : zn.getGraph().vertexSet()) { // all enabled by design
-                addVoltageControls(voltageControls, zb);
+                addMainVoltageControls(voltageControls, zb);
             }
         } else {
-            addVoltageControls(voltageControls, bus);
+            addMainVoltageControls(voltageControls, bus);
         }
         voltageControls.sort(Comparator.comparingInt(VoltageControl::getPriority));
         return voltageControls;
     }
 
     /**
+     * Check is the merged voltage to which this voltage control belongs is hidden by another one of a different type
+     * (generator, transformer or shunt). The hidden status includes the disable status so a disable voltage control is
+     * also hidden.
+     *
      * FIXME: take into account controllers status to have a proper definition
      * For generator voltage control, isGeneratorVoltageControlEnabled() should be called.
      * For transformer voltage control, isVoltageControlEnabled() should be called.
      * For shunt voltage control, isVoltageControlEnabled() should be called.
      */
     public boolean isHidden() {
-        // collect all voltage controls with the same controlled bus as this one and also all voltage controls coming
-        // from merged ones
-        List<VoltageControl<?>> voltageControls = findVoltageControlsSortedByPriority(controlledBus);
-        if (voltageControls.isEmpty()) {
+        // collect all main voltage controls connected the same controlled bus
+        List<VoltageControl<?>> mainVoltageControls = findMainVoltageControlsSortedByPriority(controlledBus);
+        if (mainVoltageControls.isEmpty()) {
             return true; // means all disabled
         } else {
-            // we should normally have max 3 voltage controls (one of each type) because already merged
-            return voltageControls.get(0) != this;
+            // we should normally have max 3 voltage controls (one of each type) with merge type as main
+            // in order to this method work whatever the voltage control is main or dependent we check against
+            // main voltage control of this
+            return mainVoltageControls.get(0) != this.getMainVoltageControl();
         }
     }
 
-    public boolean isDisabledAndAlsoAllItsDependentVoltageControls() {
-        // collect all voltage controls with the same controlled bus as this one and also all voltage controls coming
-        // from merged ones
-        List<VoltageControl<?>> voltageControls = findVoltageControlsSortedByPriority(controlledBus);
-        if (voltageControls.isEmpty()) {
-            return true; // means all disabled
+    public boolean isVisible() {
```

Comment:
If you don't need this method, I prefer to use not hidden. It is really clear to me. Visible is less.

Reply:
Or make it private.